x-pack/plugins/apm/server/lib/search_strategies/correlations/search_strategy.test.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+
+import { SearchStrategyDependencies } from 'src/plugins/data/server';
+
+import {
+  apmCorrelationsSearchStrategyProvider,
+  PartialSearchRequest,
+} from './search_strategy';
+
+// helper to trigger promises in the async search service
+const flushPromises = () => new Promise(setImmediate);
+
+const clientFieldCapsMock = () => ({ body: { fields: [] } });
+
+// minimal client mock to fulfill search requirements of the async search service to succeed
+const clientSearchMock = (
+  req: estypes.SearchRequest
+): { body: estypes.SearchResponse } => {
+  let aggregations:
+    | {
+        transaction_duration_percentiles: estypes.AggregationsTDigestPercentilesAggregate;
+      }
+    | {
+        transaction_duration_min: estypes.AggregationsValueAggregate;
+        transaction_duration_max: estypes.AggregationsValueAggregate;
+      }
+    | {
+        logspace_ranges: estypes.AggregationsMultiBucketAggregate<{
+          from: number;
+          doc_count: number;
+        }>;
+      }
+    | {
+        latency_ranges: estypes.AggregationsMultiBucketAggregate<{
+          doc_count: number;
+        }>;
+      }
+    | undefined;
+
+  // fetchTransactionDurationPercentiles
+  if (req?.body?.aggs?.transaction_duration_percentiles !== undefined) {
```

Comment:
nit: can we do an assertion for `req?.body?.aggs?` before these if statements so we don't have to duplicate these checks again?

Reply:
Refactored in 16ef560.